### ISSUE TITLE
Follow-ups from #5498: valid() permanently disables module-owned training flags

### ISSUE THREAD

**Issue by antimora** [tracel-ai/burn#5524]
Follow-ups from #5498, which generalized `Param` with module-owned flags. Full detail and repros in https://github.com/tracel-ai/burn/pull/5498#issuecomment-5457074772.

### Behavior

`Param<Flag>::valid()` writes `Flag(false)` into the value and keeps the real state in `is_active`, which is `pub(crate)` and read by nothing on any forward path. `to_device` and `fork` are the identity for flags, so only `train()` restores.

```
after valid():             flag=false
after valid().fork(ad):    flag=false
after valid().to_device(): flag=false
after valid().train():     flag=true
```

`Learner::fit` returns `model.valid()` and tensors are backend erased, so resuming a finetune from a `LearningResult` without `train()` trains with dropout off, noise off, RRelu on the fixed slope, and batch norm on frozen running statistics, silently.

- [ ] Decide the fix: either stop mutating the flag value in `valid()` and rely on the existing `is_autodiff()` guard, which already fires on all four forward paths, or make `is_active` public and document `train()` as the only restore path.
- [ ] `valid()` is no longer a no-op on a plain device module. `valid_on_a_plain_device_module_is_no_op` (`module/base.rs:643`) passes only because `SimpleLinear` has no flag.
- [ ] Decide whether `apply_lora` / `apply_qlora` should disable flags. They freeze every base weight but run through `ApplyReparameterization`, which only implements `map_float`, so the two "freeze everything" paths disagree.
- [ ] `list_param_ids` (`param/visitor.rs:12`) omits flags while `ParamGroup`'s collector includes them, so `ParamGroup::from_ids(list_param_ids(&m))` freezes tensors but no flags.

### Display

- [ ] `BatchNorm::custom_content` (`norm/batch.rs:215`) never adds `training`, so a frozen batch norm prints identically to a trainable one. Add a helper on `Param<Flag>` rather than a fourth copy of the same match.

### Docs

- [ ] `no_grad` (`module/base.rs:144`) excludes batch norm from its list of affected layers, though it is affected and changes the most.
- [ ] `freeze_group` / `unfreeze_group` (`base.rs:166`, `:180`) still describe pure `require_grad` operations.
- [ ] `burn-book/src/building-blocks/module.md:58` maps `no_grad()` onto PyTorch `require_grad_(False)`; it is now that plus `.eval()`.
- [ ] `map_flag`'s doc cites "a device move" as a mapper with no opinion. `to_device` is generated per field by the derive and never uses `ModuleMapper`.

### Tests

- [ ] `unfreeze_group`'s `training = true` arm is unexercised. Setting `base.rs:189` to `training = false` leaves the suite green while inverting the only recovery path from `no_grad()`.
- [ ] No test proves a path or predicate group can select a flag. Existing coverage uses id matching, which ignores paths, or a non-match that passes vacuously.
- [ ] The new `self.is_active = require_grad;` on the lazy branch of `set_require_grad` (`param/base.rs:564`) is unreached; needs a param that has not been initialized yet.
- [ ] `parameter_capabilities_are_split_by_value_kind` cannot fail while the build succeeds, and never asserts that `Flag` is not a `Parameter`. Needs a `compile_fail` case or removal.

**Comment by antimora** [comment]
CC @nathanielsimard @laggui

**Comment by nathanielsimard** [comment]
@laggui is the PR https://github.com/tracel-ai/burn/pull/5520 supposed to fix the issue.

**Comment by laggui** [comment]
> [@laggui](https://github.com/laggui) is the PR [#5520](https://github.com/tracel-ai/burn/pull/5520) supposed to fix the issue.

Nope those are totally unrelated. That PR is a refactor for the `burn-dispatch` declarative macros, to unify the logic with the `#[backend_extension]` proc macro since both are supposed to handle dispatch. That and a cleaner autodiff context handling.

But I'll go into the flagged points and address as needed.
